javalib/src/main/scala/java/nio/file/Files.scala

```diff
@@ -315,19 +347,25 @@ object Files {
   def lines(path: Path, cs: Charset): Stream[String] =
     newBufferedReader(path, cs).lines(true)
 
-  private def _list(dir: Path): SStream[Path] =
-    dir.toFile().list().toStream.map(dir.resolve)
-
   def list(dir: Path): Stream[Path] =
-    if (!isDirectory(dir, Array.empty)) {
-      throw new NotDirectoryException(dir.toString)
-    } else {
-      new WrappedScalaStream(_list(dir), None)
-    }
+    new WrappedScalaStream(
+      FileHelpers.list(dir.toString, (n, _) => dir.resolve(n)).toStream,
+      None)
 
   def move(source: Path, target: Path, options: Array[CopyOption]): Path = {
-    copy(source, target, options)
-    delete(source)
+    if (!exists(source.toAbsolutePath, Array.empty))
+      throw new NoSuchFileException(source.toString)
+    if (!exists(target.toAbsolutePath, Array.empty) || options.contains(
```

Comment:
Curlies around the first then clause, and then `else if` afterwards would read better.

Reply:
Done.